storage/innobase/srv/srv0start.cc

```diff
@@ -2591,7 +2591,11 @@ void innodb_shutdown()
 	srv_shutdown_all_bg_threads();
 
 	if (srv_monitor_file) {
-		fclose(srv_monitor_file);
+		if (srv_innodb_status) {
+			fclose(srv_monitor_file);
+		} else {
+			my_fclose(srv_monitor_file, MYF(0));
+		}
```

Comment:
Could we replace the `fopen()` with `my_fopen()`, so that the file would always be closed by `my_fclose()`?

Reply:
probably.